### ISSUE TITLE
Add forwarding tracking to channel state

### ISSUE THREAD

**Issue by cdecker** [ElementsProject/lightning#1786]
We currently don't keep any stats on the number of payments we forwarded, or how many fees we accumulated over time.

## `channel_metrics` table
I propose we add a new table `channel_metrics` that keeps simple counters, gauges and other numeric data that is associated with a channel. The goal is to keep the `channels` table simple while still allowing us to extend the metrics in the future. Since we have been doing this for other tables we can just use an `enum` in the source to differentiate types of metrics, and keep the footprint small:

```
channel_id: foreign_key u64 | metric_type: u32 | value: u64
```

## Updating forwarding stats
The update to these values is rather simple, since we just bump the respective number and save it alongside the channel.

**Comment by cdecker** [comment]
I decided against the separate table for now since there is a one-to-one relationship between channel and each stats type, making the existing format the 3NF :wink:

**Comment by cdecker** [comment]
Fixed by #2033
